core/src/main/scripts/importer/validateData.py

```diff
@@ -1228,6 +1232,409 @@ def checkLine(self, data):
+                log_message = "All Values in columns Reference_Allele, Tumor_Seq_Allele1 " \
+                              "and Tumor_Seq_Allele2 are equal."
+                if not self.strict_mode:
+                    self.logger.warning(log_message, extra={'line_number': self.line_number,
+                                                            'cause': '(%s, %s, %s)' % (ref_allele, tumor_seq_allele1,
+                                                                                       tumor_seq_allele2)})
+                else:
+                    self.logger.error(log_message, extra={'line_number': self.line_number,
+                                                          'cause': '(%s, %s, %s)' % (ref_allele, tumor_seq_allele1, 
+                                                                                     tumor_seq_allele2)})
+
+            # Check in case of insertion if the Reference_Allele contains '-'
+            if variant_type == "INS" and '-' not in ref_allele:
+                log_message = "Variant_Type indicates an insertion, but Reference_Allele does not equal -."
+                if not self.strict_mode:
+                    self.logger.warning(log_message, extra={'line_number': self.line_number, 'cause': ref_allele})
+                else:
+                    self.logger.error(log_message, extra={'line_number': self.line_number, 'cause': ref_allele})
+
+            # In case of deletion, check when Reference_Allele is the same length as both Tumor_Seq_Allele if at least
+            # one of the Tumor_Seq_Alleles is a deletion ('-') otherwise a SNP
+            if variant_type == "DEL" and len(ref_allele) == len(tumor_seq_allele1) \
+                    and len(ref_allele) == len(tumor_seq_allele2) and "-" not in tumor_seq_allele1 \
+                    and "-" not in tumor_seq_allele2:
+                log_message = "Variant_Type indicates a deletion, Allele based columns are the same length, " \
+                              "but Tumor_Seq_Allele columns do not contain -, indicating a SNP."
+                if not self.strict_mode:
+                    self.logger.warning(log_message, extra={'line_number': self.line_number,
+                                                            'cause': '(%s, %s, %s)' % (ref_allele, tumor_seq_allele1, 
+                                                                                       tumor_seq_allele2)})
+                else:
+                    self.logger.error(log_message, extra={'line_number': self.line_number, 
+                                                          'cause': '(%s, %s, %s)' % (ref_allele, tumor_seq_allele1, 
+                                                                                     tumor_seq_allele2)})
+
+            # In case of ONP there is checked if the Allele based columns are bigger than 3 by the MAF file checks
+            # but not if they are all the same length.
```

Comment:
This should be part of Maf file check 11f, right?

Reply:
If you'd like to keep this separate, please improve the grammar in this comment.

Reply:
I misread MAF file check 11F, it is indeed in there, i will move it there.